android/build.gradle

```diff
@@ -21,12 +21,12 @@ allprojects {
     }
 
     configurations.all {
-        exclude group: 'com.google.guava', module: 'failureaccess'
+       exclude group: 'com.google.guava', module: 'failureaccess'
 
         resolutionStrategy {
             eachDependency { details ->
                 if('guava' == details.requested.name) {
-                    details.useVersion '27.0-android'
+                    details.useVersion '31.0-android'
```

Comment:
can you explain it?

Reply:
When trying to record a video the camera api calls the method `buildOrThrow()`  resulting in this exception 

> Unhandled Exception: PlatformException(NoSuchMethodError, java.lang.NoSuchMethodError: No virtual method buildOrThrow()Lcom/google/common/collect/ImmutableMap; in class Lcom/google/common/collect/ImmutableMap$Builder

from `com/google/common/collect/ImmutableMap` which was added in 
[release v31.0](https://github.com/google/guava/releases/tag/v31.0):

> collect: Renamed ImmutableMap.Builder.build() to buildOrThrow(). The existing build() method will continue to exist but may be deprecated, and the new name should be used in new code. (https://github.com/google/guava/commit/4bbe12c4e031b6c18074f933e444fa20a371d633)

commit: https://github.com/google/guava/commit/4bbe12c4e031b6c18074f933e444fa20a371d633